comp/forwarder/orchestrator/orchestratorimpl/forwarder_orchestrator.go

```diff
@@ -0,0 +1,59 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2023-present Datadog, Inc.
+
+//go:build orchestrator
+
+// Package orchestratorimpl implements the orchestrator forwarder component.
+package orchestratorimpl
+
+import (
+	"go.uber.org/fx"
+
+	"github.com/DataDog/datadog-agent/comp/core/config"
+	"github.com/DataDog/datadog-agent/comp/core/log"
+	"github.com/DataDog/datadog-agent/comp/forwarder/defaultforwarder"
+	"github.com/DataDog/datadog-agent/comp/forwarder/orchestrator"
+	"github.com/DataDog/datadog-agent/pkg/config/resolver"
+	orchestratorconfig "github.com/DataDog/datadog-agent/pkg/orchestrator/config"
+	apicfg "github.com/DataDog/datadog-agent/pkg/process/util/api/config"
+	"github.com/DataDog/datadog-agent/pkg/util/fxutil"
+	"github.com/DataDog/datadog-agent/pkg/util/optional"
+)
+
+// Module defines the fx options for this component.
+var Module = fxutil.Component(
+	fx.Provide(newOrchestratorForwarder),
+)
+
+// newOrchestratorForwarder returns an orchestratorForwarder
+// if the feature is activated on the cluster-agent/cluster-check runner, nil otherwise
+func newOrchestratorForwarder(log log.Component, config config.Component, params Params) orchestrator.Component {
```

Comment:
should we use a `dependencies` struct like we do for other comp ?

Reply:
I can, but I think both approaches are valid. What do you think?